### PR TITLE
* Compile str.upper() and str.is...() functions

### PR DESCRIPTION
**Pull request by rubidge**
## Motivation and Context
* New feature: Compile str.upper()
* New feature: Compile str.isalpha(), str.isalnum(), str.isdecimal(), str.isdigit(), str.islower(),
    str.isnumeric(), str.isprintable, str.isspace(), str.istitle(), and str.isupper().  
    These are all the str.is...() functions, except for str.isidentifier().
   
## Approach
* str.upper() implemented the same way as str.lower()
* The str.is...() functions require a Unicode property table, which is generated by unicodeprops.py
      * Should this script be invoked in Makefile? or in setup.py?  It's currently invoked in Makefile.
* Implemented C++ functions String -> bool, for each of the standard str.is...() functions.
* Defined corresponding runtime functions to call from python, and create the appropriate call in String wrapper. 

## How Has This Been Tested?
* Created test cases for each new function.  Important cases are null strings, \x00, all codepoint sizes, titlecase codepoints, combinations of properties within string.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by rubidge** [comment]
Didn't invoke unicodeprops.py, so unicode properties header file not created, so c++ compilation failed.

File: nativepython/type_wrappers/string_wrapper.py [hunk 1]
```diff
@@ -165,6 +167,116 @@ def convert_method_call(self, context, instance, methodname, args, kwargs):
                         ).cast(self.layoutType)
                     )
                 )
+        elif methodname == "upper":
```
Comment (braxtonmckee) [inline nativepython/type_wrappers/string_wrapper.py]: might make sense to have a dictionary from method to the runtime function if the pattern is consistent

File: Makefile [hunk 2]
```diff
@@ -31,6 +32,7 @@ O_FILES = $(BUILD_PATH)/_types.o \
 install: $(VIRTUAL_ENV)
 	. $(VIRTUAL_ENV)/bin/activate; \
 		pip install pipenv==2018.11.26; \
+		$(PYTHON) $(SRC_PATH)/unicodeprops.py > $(UNICODEPROPS); \
```
Comment (braxtonmckee) [inline Makefile]: Generating UnicodeProps.hpp will either have to be part of setup.py, or done as a step where we check the file into source control (and then we just need a 'make' target to update it periodically).

Comment (rubidge) [reply Makefile]: Thanks, will do.

File: nativepython/type_wrappers/string_wrapper.py (hunk 1 above)
Comment (rubidge) [reply nativepython/type_wrappers/string_wrapper.py]: Yes, I agree.  Will do.

**Pull request closed without merging**